### PR TITLE
DOP-2215: Domain overhaul + remove all guides content

### PR DESCRIPTION
**Pull request by mango-db**
[DOP-2215](https://jira.mongodb.org/browse/DOP-2215).
[snooty DOP-2215 PR](https://github.com/mongodb/snooty/pull/427).

- Move the following new PLP directives from `landing` to `mongodb` domain: 
  - `button`
  - `kicker`
  - `procedure`
  - `step`
- ... as well as the following pre-existing [`docs-landing` directives](https://github.com/mongodb/docs-landing/blob/master/source/index.txt):
  - `introduction`
  - `card-group`
  - `card`
  - `cta`
  -  `deprecated-version-selector`
- Add `test_mongodb_domain/` to test new `mongodb` domain directives
- Deprecate `uriwriter`
- Remove `landing` domain
- Remove most `guides`-domain content (and deprecate others); we anticipate @rayangler will handle next-gen guides 🥳
   - This is being done in this PR because of the `procedure` directive conflicts
- In `rstspec.toml`:
   - Group `mongodb`-domain **directives** together (including `mongowebshell`, `openapi`, and `operation`, which are otherwise irrelevant to this change)
   - Group `mongodb`-domain **roles** together (`ref` and ` card-ref` (which must inherit from `std`-domain `ref`), as well as the`required` role)
 
Notes:
- Breaks [BI-connector cards](https://github.com/mongodb/docs-bi-connector/blob/4460b478bc2f68dc4bc2c52968bf54ce0e36596d/source/client-applications.txt#L32) used [only on this page](https://docs.mongodb.com/bi-connector/current/client-applications/), which are being deprecated. This is handled by [this PR](https://github.com/mongodb/docs-bi-connector/pull/313).
- Decision was made with team approval to remove the `landing` domain entirely.
- Decision was made with team approval to remove the stale `guides` content
   - This PR further breaks docs-tutorials builds, which is acceptable because we plan to fully re-implement it in the future
- Decision was made to distinguish rST versus YAML-generated steps by naming YAML-generated steps `step-yaml` in the AST, which is being handled in a separate PR

## Related PRs:
- [React frontend](https://github.com/mongodb/snooty/pull/427) change
- [BI-connector](https://github.com/mongodb/docs-bi-connector/pull/313) card change
- [Docs-landing](https://github.com/mongodb/docs-landing/pull/75) default_domain change

#### YAML-generated step functionality proof
[Realm](https://docs-mongodbcom-staging.corp.mongodb.com/master/realm/cgoecknerwald/master/sdk/ios/install/), [rST](https://github.com/mongodb/docs-realm/blob/master/source/sdk/ios/install.txt#L50)

### CODE REVIEW

File: snooty/rstspec.toml
Comment (schmalliso) [inline snooty/rstspec.toml:653]: We've now got mongodb:card-group and also [directive.card-group](https://github.com/mongodb/snooty-parser/pull/314/files#diff-8c18bfff17bea8d40e6b0338264e6176ec2b78618336920fa42b71c5409dd88fR458) on line 458 -- do we want/need both?

File: snooty/test_parser.py
Comment (schmalliso) [inline snooty/test_parser.py:1974]: This content format is used in the BI connector docs on the [Connect Client Applications page](https://docs.mongodb.com/bi-connector/current/client-applications/) ([source](https://github.com/mongodb/docs-bi-connector/blob/master/source/client-applications.txt#L32-L75)). I'm fine deprecating this if we provide them with an alternative and send in a PR with any relevant content changes to use the alternative solution.

File: snooty/rstspec.toml
Comment (schmalliso) [inline snooty/rstspec.toml:690]: is `cta` the only directive left in the `landing` domain -- if so, what's the rationale behind keeping it in that domain rather than moving it to `mongodb`?

**Review by schmalliso — COMMENTED** [review]
Left some questions, I'll defer to Raymund and/or Andrew for a more thorough review :)

File: snooty/rstspec.toml
Comment (rayangler) [inline snooty/rstspec.toml:709]: Would this change affect the way steps are generated from `steps-*.yaml` files? Would moving this `step` directive out of the `landing` domain conflict with the current implementation of the frontend `Step` [component](https://github.com/mongodb/snooty/blob/master/src/components/Step.js) and its use in any of the [guides](https://docs.mongodb.com/guides/) (after nextgen) or other docs pages like [this](https://docs.mongodb.com/realm/sdk/ios/install/#installation)?

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:653]: Good question - I left in `directive.card-group` because it is [in-use](https://github.com/mongodb/docs-bi-connector/blob/4460b478bc2f68dc4bc2c52968bf54ce0e36596d/source/client-applications.txt#L32) by `docs-bi-connector`. However, because the `mongodb` domain is searched first, and because we have a defined `mongodb:card-group`, `docs-bi-connector` never finds `std:card-group` when parsed, and just gives a warning about the specified `:type:` option (which is in the `std:card-group` but not the `mongodb:card-group`). So, I can remove this line - but it might be a good idea to figure out a way to more gracefully transition the `bi-connector` cards to the new `card-group` & `card` type

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:690]: Yes! That and `deprecated-version-selector`, which I'm not sure of the importance of (I only see it used [here](https://github.com/mongodb/docs-landing/blob/master/source/legacy.txt#L14)). I'm generally leaning towards getting rid of the landing domain, since it's so sparse

**Review by rayangler — COMMENTED** [review]
> These changes leave the landing domain with only `cta` and `deprecated-version-selector`.

I think it should be fine to move these out of the `landing` domain. Although `deprecated-version-selector` is technically only reserved for [legacy](https://docs.mongodb.com/legacy/) on the landing site

> I am unsure of how to smoothly handle transition to new `card`s

What do you think about either turning BI-Connector's cards to a new type of card directive like `card-image` (name pending), or maybe even add a new field to the existing landing `card`, or even on `card-group`, that changes the type or variant of card we want to render?

File: snooty/legacy_guides.py
Comment (mango-db) [reply snooty/legacy_guides.py:172]: Probably @schmalliso and @i80and ^^

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:649]: Removing `landing` domain...

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:686]: Q1. This change would not affect the way steps are generated from `steps-*.yaml` files. YAML in general is handled by `/gizaparser`; steps in particular are handled in [parser.py](https://github.com/mongodb/snooty-parser/blob/master/snooty/parser.py#L1261-L1281) and [gizaparser/steps.py](https://github.com/mongodb/snooty-parser/blob/master/snooty/gizaparser/steps.py) [(test_steps.py)](https://github.com/mongodb/snooty-parser/blob/master/snooty/gizaparser/test_steps.py). 

That being said, this PR currently generally adds breakage to `docs-tutorials`, which is acceptable because we are overhauling it.

Added the following staging for YAML `step` sanity check:
[Realm](https://docs-mongodbcom-staging.corp.mongodb.com/master/realm/cgoecknerwald/master/sdk/ios/install/)

File: snooty/legacy_guides.py
Comment (mango-db) [reply snooty/legacy_guides.py:172]: `snooty/legacy_guides.py` has been removed.

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:687]: Q2: Moving this step out of the `landing` domain does not conflict with the current implementation of the frontend `Step` component because the parser output of `.. include:: steps-*.yaml` does not change (handled by `gizaparser`).

### Input rST/YAML layer
Even though we now have a `step` directive in a searched domain, the parser differentiates these two step input-formats. (You can see a sample JSON output [here](https://github.com/mongodb/snooty/blob/master/tests/unit/data/Step.test.json).)

### Output AST layer
One might also be concerned that a parsed `.. step::` directive may end up being accidentally rendered as the original YAML `Step` component. There are will be no usages of the `.. step::` directive until the PLP feature branch is merged, so we do not have to worry about conflicts in the interim.

### PLP consequences
We can add a test for `domain = "mongodb"` in `ComponentFactory` to distinguish the two `step` directives for now. This is analogous to how the `step` directive is currently functional (within the `landing` domain). Alternately, we could rename the gizaparser output to `step-yaml`, which I am in favor of.

Because we now have a proper `.. step::` directive, and because we can anticipate moving [PLP's landing-domain Step component](https://github.com/mongodb/snooty/blob/plp/src/components/landing/Step.js) out of the `components/landing/` subfolder, there are liable to be naming conflicts/confusions with the existing [components/Step.js](https://github.com/mongodb/snooty/blob/master/src/components/Step.js), [components/Steps.js](https://github.com/mongodb/snooty/blob/master/src/components/Steps.js), and [components/Stepper.js](https://github.com/mongodb/snooty/blob/master/src/components/Stepper.js). I am mostly likely going to propose renaming existing Step rendering to YAML-Step to differentiate.

The PLP feature branch will handle this update in [this PR](https://github.com/mongodb/snooty/pull/427).

**Review by schmalliso — APPROVED** [review]

File: snooty/test_parser.py
Comment (mango-db) [reply snooty/test_parser.py:1974]: Yes! The new `card-group`/`card` directives are tested in `test_mongodb_domain.py`

File: snooty/rstspec.toml
Comment (mango-db) [reply snooty/rstspec.toml:687]: YAML step name-conflict issues handled by [DOP-2249](https://jira.mongodb.org/browse/DOP-2249).

**Review by i80and — APPROVED** [review]
happyhappy

LOVE IT

**Comment by schmalliso** [comment]
All the best PRs remove more code than they add. :swoon:

**Pull request merged**
